crates/swc_ecma_minifier/src/analyzer/storage/normal.rs

```diff
@@ -153,32 +153,27 @@ impl Storage for ProgramData {
         //     debug!(has_init = has_init, "declare_decl(`{}`)", i);
         // }
 
-        let v = self
-            .vars
-            .entry(i.to_id())
-            .and_modify(|v| {
-                if has_init && (v.declared || v.var_initialized) {
-                    trace_op!("declare_decl(`{}`): Already declared", i);
-
-                    v.mutated = true;
-                    v.reassigned_with_var_decl = true;
-                    v.assign_count += 1;
-                }
+        let v = self.vars.entry(i.to_id()).or_default();
 
-                if v.used_in_non_child_fn {
-                    v.is_fn_local = false;
-                }
+        if has_init && (v.declared || v.var_initialized) {
+            trace_op!("declare_decl(`{}`): Already declared", i);
 
-                v.var_initialized |= has_init;
-            })
-            .or_insert_with(|| VarUsageInfo {
```

Comment:
This is problematic because this closure is not called if the analyzer called `.var_or_default()` for the infection analysis